sig/transport/request_options.rbs

```diff
@@ -0,0 +1,33 @@
+module Algolia
+  module Transport
+    class RequestOptions
+      attr_accessor headers: Hash[Symbol|String, String]
+
+      attr_accessor params: Hash[Symbol|String, untyped]
+
+      attr_accessor data: Hash[Symbol|String, untyped]
+
+      attr_accessor timeout: Integer
+
+      attr_accessor connect_timeout: Integer
+
+      attr_accessor compression_type: String
+
+      def initialize: (Algolia::Search::Config config) -> void
```

Comment:
I don't think it's necessary to prefix with `Algolia::`

Reply:
It's not, indeed. However, to prevent any namespace clashes, I personally write out the full namespace all the time. I believe other libraries (like Elastic's Ruby client) do the same thing. Wdyt, should we leave it, or remove it?

Reply:
okay let's keep it!